logstash-core/src/main/java/org/logstash/config/ir/compiler/Dataset.java

```diff
@@ -0,0 +1,438 @@
+package org.logstash.config.ir.compiler;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import org.jruby.RubyArray;
+import org.jruby.RubyHash;
+import org.logstash.ext.JrubyEventExtLibrary;
+
+/**
+ * A data structure backed by a {@link RubyArray} that is lazily filled with
```

Comment:
@jordansissel right sorry about that weird doc, tough week(s) it seems :D Will add the first half of whatever doc I was planning here :)